### PR TITLE
T0895 mg tostring function should accept date data type as argument

### PR DESCRIPTION
**Pull request by gvolfing**
[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

### CODE REVIEW

File: src/utils/temporal.cpp [hunk 1]
```diff
@@ -486,6 +500,8 @@ int64_t LocalDateTime::SubSecondsAsNanoseconds() const {
   return (milli_as_nanos + micros_as_nanos).count();
 }
 
+std::string LocalDateTime::ToString() const { return date.ToString() + "T" + local_time.ToString(); }
```
Comment (antaljanosbenjamin) [inline src/utils/temporal.cpp:503]:
```suggestion
std::string LocalDateTime::ToString() const { return date.ToString() + 'T' + local_time.ToString(); }
```

File: tests/unit/utils_temporal.cpp [hunk 2]
```diff
@@ -609,3 +609,61 @@ TEST(TemporalTest, LocalDateTimeDelta) {
...
+
+TEST(TemporalTest, DurationConvertsToString) {
+  memgraph::utils::Duration duration1{{.minute = 2, .second = 2, .microsecond = 33}};
+  const std::string duration1_expected_str = "P0DT0H2M2.000033S";
+  memgraph::utils::Duration duration2{{.hour = 2.5, .minute = 2, .second = 2, .microsecond = 33}};
+  const std::string duration2_expected_str = "P0DT2H32M2.000033S";
+  memgraph::utils::Duration duration3{{.hour = 1.25, .minute = 2, .second = 2}};
+  const std::string duration3_expected_str = "P0DT1H17M2.000000S";
+  memgraph::utils::Duration duration4{{.day = 20, .hour = 1.25, .minute = 2, .second = 2}};
+  const std::string duration4_expected_str = "P20DT1H17M2.000000S";
```
Comment (antaljanosbenjamin) [inline tests/unit/utils_temporal.cpp:663]: Maybe modify/add a test case to test negative values.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

File: tests/unit/utils_temporal.cpp (hunk 2 above)
Comment (jbajic) [reply tests/unit/utils_temporal.cpp:663]: Also add cases where all units need to overflow into upper, e.g. 
`memgraph::utils::Duration duration{{.day = 20, .hour = 72, .minute = 154, .second = 312}};`

File: src/utils/temporal.cpp [hunk 3]
```diff
@@ -175,6 +176,10 @@ int64_t Date::MicrosecondsSinceEpoch() const {
 
 int64_t Date::DaysSinceEpoch() const { return utils::DaysSinceEpoch(year, month, day).count(); }
 
+std::string Date::ToString() const {
+  return fmt::format("{:0>4}-{:0>2}-{:0>2}", year, static_cast<int>(month), static_cast<int>(day));
```
[the remark this answers is not in the document]
Comment (gvolfing) [reply src/utils/temporal.cpp:180]: This was copied from the previous implementation of `operator<<`. But I am guessing because `month` and `day` are declared as `uint8_t` which is formatted as a character by `fmt::format()`

[the remark this answers is not in the document]
Comment (antaljanosbenjamin) [reply src/utils/temporal.cpp:180]: Yes, can confirm.

**Review by jbajic — APPROVED** [review]

**Review by antaljanosbenjamin — APPROVED** [review]
